README.md

```diff
@@ -830,223 +815,155 @@ With multiversion support, you can have two members where each of them having di
 };
-```
 
-Similar to `IdentifiedDataSerializable`, a `Portable` object must provide `classId` and `factoryId`. The factory object will be used to create the `Portable` object given the `classId`.
+namespace hazelcast {
+    namespace client {
+        namespace serialization {
+            template<>
+            struct hz_serializer<Person> : portable_serializer {
+                static int32_t get_factory_id() noexcept {
+                    return 1;
+                }
 
-A sample `PortableFactory` could be implemented as follows:
+                static int32_t get_class_id() noexcept {
+                    return 3;
+                }
 
-```C++
-class SamplePortableFactory : public serialization::PortableFactory {
-public:
-    static const int FACTORY_ID = 1;
-
-    virtual std::auto_ptr<serialization::Portable> create(int32_t classId) const {
-        switch (classId) {
-            case 1:
-                return std::auto_ptr<serialization::Portable>(new PortableSerializableSample());
-            default:
-                return std::auto_ptr<serialization::Portable>();
+                static void write_portable(const Person &object, hazelcast::client::serialization::portable_writer &out) {
+                    out.write("name", object.name);
+                    out.write("gender", object.male);
+                    out.write("age", object.age);
+                }
+
+                static Person read_portable(hazelcast::client::serialization::portable_reader &in) {
+                    return Person{in.read<std::string>("name"), in.read<bool>("gender"), in.read<int32_t>("age")};
+                }
+            };
         }
     }
-};
-```
-
-The last step is to register the `PortableFactory` to the `serialization_config`.
+}
 
-```C++
-    client_config clientConfig;
-    clientConfig.get_serialization_config().addPortableFactory(SamplePortableFactory::FACTORY_ID,
-                                                             boost::shared_ptr<serialization::PortableFactory>(
-                                                                     new SamplePortableFactory()));
 ```
 
-Note that the ID that is passed to the `serialization_config` is same as the `factoryId` that `PortableSerializableSample` object returns.
+Note that, this is very similar to `identified_data_serializer` implementation except the method signatures and the derived marker class of `portable_serializer` for the specialized serializer.
+
+`hz_serializer<Person>` specialization of `hz_serializer` should implement the above four methods, namely `get_factory_id`, `get_class_id`, `write_portable`, `read_portable`. In case that the object fields are non-public, you can always define struct `hz_serializer<Person>` as friend to your object class. You use the `portable_writer` class methods when serializing the object into binary bytes and you use the `portable_reader` class methods when de-serializing the bytes into the concrete object instance. 
```

Comment:
... as a friend to your ....

... into binary bytes and the `portable_reader` class ...